core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1028,6 +1029,10 @@ object SnappyContext extends Logging {
     }
   }
 
+  private def updateUI(sc: SparkContext): Unit = {
+    ToolsCallbackInit.toolsCallback.updateUI(sc.ui)
+  }
+
   private def invokeServices(sc: SparkContext): Unit = {
     SnappyContext.getClusterMode(sc) match {
       case SnappyEmbeddedMode(_, _) =>
```

Comment:
You should also think, how new pages can be made pluggable. If required propose a change request upstream Spark. So that we don't have keep everything in Spark.